src/extension_utils.ts

```diff
@@ -12,14 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 import * as yaml from 'yaml';
-import { readFileSync } from 'fs';
+import { satisfies, minVersion, parse } from 'semver';
+import { isString } from 'util';
+
+const dependencyKey = 'dependencies';
+const devDependencyKey = 'dev_dependencies';
+
+function getDependencyVersion(document:yaml.Document, dependency:string, isADevDependency:boolean = false):string|undefined {
+    const key = isADevDependency ? devDependencyKey : dependencyKey;
+    if (document.contents.has(key)) {
+        const dependencyCollection = document.contents.get(key);
+        if (dependencyCollection !== null && dependencyCollection.has(dependency)) {
+            const collectedDependency = dependencyCollection.get(dependency);
+    
+            // Simple key value dependencies (e.g. not hosted ones) are simple
+            // strings here. Otherwise, we need to do `get` one more time.
+            if (isString(collectedDependency)) {
+                return collectedDependency;
+            } else {
+                return collectedDependency.get('version');
+            }
+        }
+    }
+
+    return undefined;
+}
+
+// Verifies that a provided dependency is present and has a minimum version that matches a given dependency range.
+//
+// Returns false if the dependency does not exist, is invalid, or is below the provided range.
+// Expects `dependency` to be the exact dependency string, `qualifyingRange` to be a compatible semver range.
+export function dependencyHasValidMinVersion(dependency:string, qualifyingRange:string, pubspecContents:string, isADevDependency:boolean = false):Boolean {
```

Comment:
Formatting is pretty gross at times. I started to work on it by hand and realized that really, we just need to add prettier and run the formatter. That would have made this diff horrid though, so for now I just kept the gross formatting.